### PR TITLE
Mod apbs

### PR DESCRIPTION
**Pull request by gawells**
fixColumns now works on a copy of the selection to generate the pqr file, in order to prevent changes to existing visualisations that might happen with the previous version (I noticed this with chains being lost, I use colour by chain a lot)

### CODE REVIEW

File: plugins/apbsplugin.py [hunk 1]
```diff
@@ -486,7 +486,7 @@ def getApbsInputFile(pqr_filename,
     swin 0.3         # Surface cubic spline window .. default 0.3
     temp %f          # System temperature (298.15 default)
     sdens %f         # Specify the number of grid points per square-angstrom to use in Vacc object. Ignored when srad is 0.0 (see srad) or srfm is spl2 (see srfm). There is a direct correlation between the value used for the Vacc sphere density, the accuracy of the Vacc object, and the APBS calculation time. APBS default value is 10.0.
-    #gamma 0.105      # Surface tension parameter for apolar forces (in kJ/mol/A^2)
+    #gammpa 0.105      # Surface tension parameter for apolar forces (in kJ/mol/A^2)
```
Comment (speleo3) [inline plugins/apbsplugin.py]: was this intentional?

File: plugins/apbsplugin.py [hunk 2]
```diff
@@ -1667,10 +1687,14 @@ def show_error(message):
         # copied from WLD code
         sel = "((%s) or (neighbor (%s) and hydro))" % (
             self.selection.getvalue(), self.selection.getvalue())
+
```
Comment (speleo3) [inline plugins/apbsplugin.py]: the select/create should happen here, and fixColumns should remain unchanged

**Comment by speleo3** [comment]
You modified fixColumns to create selections/objects, but they get deleted outside the function. However, fixColumns is called in more than one place. I suggest to not modify fixColumns, but do both the creating and deleting outside the function.

Also, instead of using "apbs_sel" and "apbs_copy" as hard-coded names, I suggest to use cmd.get_unused_name(): http://pymolwiki.org/index.php/Get_unused_name

**Comment by gawells** [comment]
A redo of running fixColumns on a copied object, in response to Thomas Holder's suggestions

File: plugins/apbsplugin.py [hunk 3]
```diff
@@ -1667,10 +1667,20 @@ def show_error(message):
         # copied from WLD code
         sel = "((%s) or (neighbor (%s) and hydro))" % (
             self.selection.getvalue(), self.selection.getvalue())
-        self.fixColumns(sel)
-        pymol.cmd.save(pdb_filename, sel)
+
+        apbs_sel = pymol.cmd.get_unused_name()
+        apbs_clone = pymol.cmd.get_unused_name()
```
Comment (speleo3) [inline plugins/apbsplugin.py]: calling get_unused_name() twice will return the same name, unless the name is actually used before the second call. In this case, the named selection is redundant and not needed, simply do: `pymol.cmd.create(apbs_clone, sel)`

File: plugins/apbsplugin.py [hunk 4]
```diff
@@ -1818,7 +1828,17 @@ def _generatePymolPqrFile(self):
...
+        pymol.cmd.select(apbs_sel,sel)
+        pymol.cmd.create(apbs_clone,apbs_sel) 
+        
+        self.fixColumns(apbs_clone)
+
+        pymol.cmd.delete(apbs_sel)
+        pymol.cmd.delete(apbs_clone) 
+
         pymol.cmd.save(pqr_filename, sel)
```
Comment (speleo3) [inline plugins/apbsplugin.py]: still using `sel` here instead of `apbs_clone`

**Comment by gawells** [comment]
apbs_clone selection step removed

**Comment by gawells** [comment]
fixed sel/apbs_clone mixup

**Comment by speleo3** [comment]
squash merged into master: 4ed3bff249ea1d9603641ddea94c2d611fd46d90

**Pull request closed without merging**